modules/home-manager/fish.nix

```
@@ -0,0 +1,33 @@
+{ config
+, pkgs
+, lib
+, ...
+}:
+let
+  inherit (lib) ctp mkIf;
+  cfg = config.programs.fish.catppuccin;
+  enable = cfg.enable && config.programs.fish.enable;
+
+  theme = pkgs.fetchFromGitHub
+      {
+        owner = "catppuccin";
+        repo = "fish";
+        rev = "91e6d6721362be05a5c62e235ed8517d90c567c9";
+        sha256 = "sha256-l9V7YMfJWhKDL65dNbxaddhaM6GJ0CFZ6z+4R6MJwBA=";
+      };
+  themeName = "Catppuccin ${ctp.mkUpper cfg.flavour}";
+in
+{
+  options.programs.fish.catppuccin = lib.ctp.mkCatppuccinOpt "fish" config;
+
+  # xdg is required for this to work
+  config =
+    {
+      xdg.enable = mkIf enable (lib.mkForce true);
```

```suggestion
  config = mkIf enable
    {
      xdg.enable = lib.mkForce true;
```

(nixpkgs-fmt will probably not like this but you get the point :p)